code/datums/cargo.dm

```diff
@@ -0,0 +1,378 @@
+	var/list/categories = list() //List of categories this item appears in
+	var/list/suppliers = list() //List of supliers and their prices
+	var/amount = 1 //Amount of items in the crate
+	var/access = null //What access requirement should be added to the container
+	var/container_type = "crate" //crate or box
+	var/groupable = 1 //If the item can be thrown into the same container as other items
+
+//Gets a list of the cargo item - To be json encoded
+/datum/cargo_item/proc/get_list()
+	var/list/data = list()
+	data["name"] = name
+	data["path"] = path
+	data["description"] = description
+	data["categories"] = categories
+	data["amount"] = amount
+	var/suppliers_list = list()
+	for(var/supplier in suppliers)
+		var/list/sd = list()
+		var/datum/cargo_supplier/cs = suppliers[supplier]["supplier_datum"]
+		if(!cs)
+			log_debug("Cargo - Invalid supplier [supplier] encountered when loading item [name] with path [path]")
+			continue
+		sd["short_name"] = cs.short_name
+		sd["name"] = cs.name
+		sd["available"] = cs.available
+		sd["price_modifier"] = cs.price_modifier
+		sd["base_purchase_price"] = suppliers[supplier]["base_purchase_price"]
+		sd["vars"] = suppliers[supplier]["vars"]
+		sd["adjusted_price"] = sd["base_purchase_price"] * sd["price_modifier"]
+		//Add category adjustments
+		for(var/category in categories)
+			var/datum/cargo_category/cc = SScargo.get_category_by_name(category)
+			if(cc)
+				sd["adjusted_price"] *= cc.price_modifier
+		suppliers_list[supplier] = sd
+	data["suppliers"] = suppliers_list
+	return data
+
+/*
+	A supplier of items
+*/
+/datum/cargo_supplier
+	var/short_name = "" //Short name of the cargo supplier
+	var/name = "" //Long name of the cargo supplier
+	var/description = "" //Description of the supplier
+	var/tag_line = "" //Tag line of the supplier
+	var/shuttle_time = 0 //Time the shuttle takes to get to the supplier
+	var/shuttle_price = 0 //Price to call the shuttle
+	var/available = 1 //If the supplier is available
+	var/price_modifier = 1 //Price modifier for the supplier
+//Gets a list of supplier - to be json encoded
+/datum/cargo_supplier/proc/get_list()
+	var/list/data = list()
+	data["short_name"] = short_name
+	data["name"] = name
+	data["description"] = description
+	data["tag_line"] = tag_line
+	data["shuttle_time"] = shuttle_time
+	data["shuttle_price"] = shuttle_price
+	data["available"] = available
+	data["price_modifier"] = price_modifier
+	return data
+
+/*
+	A category displayed in the cargo order app
+*/
+/datum/cargo_category
+	var/name = "cargo_category" //Name of the category
+	var/display_name = "Cargo Category"
+	var/description = "You should not see this" //Description of the Category
+	var/icon = "gear" //NanoUI Icon for the category
+	var/price_modifier = 1 //Price Modifier for the category
+	var/list/items = list() //List of items in the category
+
+// Gets a list of the cargo category - to be json encoded
+/datum/cargo_category/proc/get_list()
+	var/list/data = list()
```

Comment:
Style questions. Its either done for every get_list() in that file or none.
It cant be done for every get list because of the cargo_item get_list()
For consistency it will be left the way it is.